src/Core/Material/SimpleMaterialModel.hpp

```diff
@@ -12,7 +15,14 @@ class RA_CORE_API SimpleMaterialModel : public MaterialModel
 {
   protected:
     SimpleMaterialModel( const std::string& name, const std::string type ) :
-        MaterialModel( name, type ) {}
+        m_generator( new Core::Random::MersenneTwisterGenerator() ), MaterialModel( name, type ) {}
+
+    SimpleMaterialModel( const std::string& name,
+                         const std::string type,
+                         Core::Random::UniformGenerator* generator ) :
+        MaterialModel( name, type ) {
+        *m_generator = *generator;
+    }
```

Comment:
Use a shared ptr to store the generator. And these two constructors could be merged into one as suggested below.
Remove the constructor SimpleMaterialModel( const std::string& name, const std::string type ) and add the following

```suggestion
    SimpleMaterialModel( const std::string& name,
                         const std::string type,
                         std::shared_ptr<Core::Random::UniformGenerator> generator = std::make_shared<Core::Random::UniformGenerator>() ) :
        MaterialModel( name, type ),
        m_generator = std::move(generator) {}
```